OpenRA.Mods.Common/Scripting/ScriptTriggers.cs

```diff
@@ -421,6 +422,48 @@ public void OnDiscovered(Actor self, Player discoverer, bool playNotification)
 			}
 		}
 
```

Comment:
The reason that the compiler complains is that you've forgotten to implement `INotifyPassengerEntered` and `INotifyPassengerExited` on `ScriptTriggers`.  This means that your triggers will never actually fire.